specs/rollup-node.md

```diff
@@ -251,6 +253,46 @@ However, they can still be challenged by a fault proof until the end of the faul
 
 > **TODO** the spec doesn't encode the notion of fault proof yet, revisit this (and include links) when it does
 
+## From L2 Block to L2 Output Root
+
+After processing a block the resulting outputs will need to be synchronized with L1 for trustless execution of
+L2-to-L1 messaging, such as withdrawals. Outputs are hashed in a tree structured form for minimal proof cost to any
+piece of data. This merkle-structure is defined with [SSZ], a type system for merkleization and serialization, used in
+L1 (beacon-chain). However, we replace `sha256` with `keccak256` to save gas costs in the EVM.
+
+[SSZ]: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md
+
+```python
+class L2Output(Container):
+  state_root: Bytes32
+  withdrawal_storage_root: Bytes32
+  latest_block: ExecutionPayload  # includes block-hash
+  history_accumulator_root: Bytes32  # Not functional yet
+  extension: Bytes32
+```
+
+The `state_root` is the Merkle-Patricia-Trie ([MPT][g-mpt]) root of all execution-layer accounts,
+also found in `latest_block.state_root`: this field is frequently used and thus elevated closer to the L2 output root,
+reducing the merkle proof depth and thus the cost of usage.
+
+The `withdrawal_storage_root` elevates the Merkle-Patricia-Trie ([MPT][g-mpt]) root of L2 Withdrawal contract storage.
+Instead of a MPT proof to the Withdrawal contract account in the account trie,
+one can directly access the MPT storage trie root, thus reducing the verification cost of withdrawals on L1.
+
+The `latest_block` is an execution-layer block of L2, represented as the [`ExecutionPayload`][ExecutionPayload] SSZ type
+defined in L1. There may be multiple blocks per L2 output root, only the latest is presented.
+
+[ExecutionPayload]: https://github.com/ethereum/consensus-specs/blob/dev/specs/bellatrix/beacon-chain.md#executionpayload
+
+The `history_accumulator_root` is a reserved field, elevating a storage variable of the L2 chain that maintains
+the SSZ merkle root of an append-only `List[Bytes32, MAX_ITEM_COUNT]` (`keccak256` SSZ),
+where each item is defined as `keccak256(l2_block_hash ++ l2_state_root)`, one per block of the L2 chain.
+While reserved, a zeroed `Bytes32` is used instead.
+This is a work-in-progress, see [issue 181](https://github.com/ethereum-optimism/optimistic-specs/issues/181).
+
+The `extension` is a zeroed `Bytes32`, to be substituted with a SSZ container to extend merkleized information in future
+upgrades. This keeps the static merkle structure forwards-compatible.
+
 ## Whole L2 Chain Derivation
 
 The previous two sections presents an inductive process: given that we know the "current" L2 block, well as the next L1
```

Comment:
No longer accurate :) Could temporarily change to the previous 3 section + see my comment about potentially splitting this spec.